### DISCUSSION TITLE
form.validate().then 里做异步操作，会清空表单

### DISCUSSION THREAD

**Discussion by TheKenS** [alibaba/formily#3370] [Q&A 2.x]
form
      .validate()
      .then(() => {
         submitRquest();  // 异步操作
      })
      .catch((e: any) => {
        console.log(e);
      });

**Comment by janryWang** [comment]
给demo

**Reply by TheKenS** [reply]
emmm，找到原因了，form.submit() 里使用 ahooks 封装的 runAsync ，就会默认进行一次 form.reset() 操作，formily 是不是跟ahooks八字不合

**Comment by lianshengxing** [comment]
请问您是如何解决的该问题，同样遇到调用validate(pattern) 方法，响应器中的$deps被清空了
